backend/common/services/simulation_queue_service.py

```
@@ -486,6 +494,47 @@ def get_queue_length() -> int:
+    Returns:
+        Dict mapping worker_id to count, e.g., {"0": 3, "1": 2}
+    """
+    try:
+        result = {}
+        keys = redis_manager.keys(IN_PROGRESS_PATTERN)
+        for key in keys:
+            # Extract worker ID from key (simulation:in_progress:{worker_id})
+            worker_id = key.split(":")[-1] if isinstance(key, str) else key.decode().split(":")[-1]
+            count = redis_manager.scard(key)
+            result[worker_id] = count
+        return result
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)